app/services/dashboard/participants.rb

```diff
@@ -114,38 +106,67 @@ def currently_training_ect(induction_record)
     end
 
-    def deferred_or_transferred_ect(induction_record)
-      @deferred_or_transferred_ects << dashboard_participant(induction_record.participant_profile_id, induction_record:)
+    def no_longer_training_ect(induction_record)
+      @no_longer_training_ects << dashboard_participant(induction_record.participant_profile_id, induction_record:)
+    end
+
+    def withdrawn_ect(induction_record)
+      @withdrawn_ects << dashboard_participant(induction_record.participant_profile_id, induction_record:)
     end
 
     def orphan_ect(induction_record)
       @orphan_ects << dashboard_participant(induction_record.participant_profile_id, induction_record:)
     end
 
+    def no_longer_training_mentor(induction_record)
+      @no_longer_training_mentors << dashboard_participant(induction_record.participant_profile_id, induction_record:)
+    end
+
     def process_participants
+      process_ects
+      process_mentors
+    end
+
+    def process_ects
       induction_records
         .select(&:ect?)
         .each do |induction_record|
           next completed_induction_ect(induction_record) if induction_record.completed_induction_status?
-          next deferred_or_transferred_ect(induction_record) if induction_record.deferred_or_transferred?
+          next no_longer_training_ect(induction_record) if induction_record.deferred_or_transferred? || induction_record.withdrawn_induction_status?
           next orphan_ect(induction_record) if induction_record.mentor_profile_id.blank?
 
           currently_training_ect(induction_record)
         end
+    end
 
+    # Discover all the relevant mentors of the school, including:
+    # - orphan mentors
+    # - mentors with mentees
+    # - mentors linked to the school's ects, but not in the school's mentor pool
+    def process_mentors
+      induction_records
+        .reject(&:ect?)
+        .each do |induction_record|
+          next no_longer_training_mentor(induction_record) if induction_record.deferred_or_transferred? || induction_record.withdrawn_induction_status?
+
+          @active_mentors << dashboard_participant(induction_record.participant_profile_id, induction_record:)
+        end
+
+      # Create a hash with the format below, to display the mentors in the "Currently training" filter
+      # { mentor_dashboard_participant => [ect_dashboard_participant_1, ect_dashboard_participant_2] }
       @mentors = @currently_training_ects.group_by(&:mentor_profile_id).transform_keys do |mentor_profile_id|
         dashboard_participant(mentor_profile_id)
       end
-    end
 
-    def school_mentors_not_mentoring
-      school.school_mentors.reject do |school_mentor|
-        profile_ids_of_mentors_mentoring.include?(school_mentor.participant_profile_id)
+      # Include also the orphan mentors in the @mentors
+      @orphan_mentor_ids = @active_mentors.map(&:participant_profile_id) - @currently_training_ects.map(&:mentor_profile_id)
+      @orphan_mentor_ids.each do |mentor_profile_id|
```

Comment:
We we need an instance variable (or even a variable) to store orphan_mentor_ids?